test_utils/basetest.py

```diff
@@ -3,100 +3,251 @@
+            collector_id = json.loads(resp.text)['collector']['id']
+            cls.logger.info("created collector {}".format(collector_name))
+        except Exception as e:
+            raise Exception(e)
+
+        return collector_id
+    
+    @classmethod
+    def delete_collector(cls, collector_id):
+        sources = cls.sumologic_cli.sources(collector_id, limit=10)
+        if len(sources) == 0:
+            cls.sumologic_cli.delete_collector(
+                {"collector": {"id": collector_id}})
+            cls.logger.info("deleted collector")
+
+    @classmethod
+    def create_source(cls, collector_id, source_name):
+        cls.logger.info("creating source")
+        endpoint = source_id = None
+        params = {
+            "sourceType": "HTTP",
+            "name": source_name,
+            "messagePerRequest": False,
+            "multilineProcessingEnabled": True,
+            "category": "AZURE/UnitTest/logs"
+        }
 
-        return repo_name, branch_name
+        try:
+            resp = cls.sumologic_cli.create_source(collector_id, {"source": params})
+            data = resp.json()['source']
+            source_id = data["id"]
+            endpoint = data["url"]
+            cls.logger.info("created source {}".format(source_name))
+        except Exception as e:
+            raise Exception(e)
+        return source_id, endpoint
+    
+    @classmethod
+    def delete_source(cls, collector_id, source_id):
+        cls.sumologic_cli.delete_source(
+            collector_id, {"source": {"id": source_id}})
+        cls.logger.info("deleted source")
+
+    def fetchlogs(self, app_insights):
+        result = []
+        try:
+            client = LogsQueryClient(self.azure_credential)
+            query = f"app('{app_insights}').traces | where operation_Name == '{self.function_name}' | project operation_Id, timestamp, message, severityLevel"
+            response = client.query_workspace(
+                self.get_Workspace_Id(), query, timespan=timedelta(hours=1))
+
+            if response.status == LogsQueryStatus.FAILURE:
+                raise Exception(f"LogsQueryError: {response.message}")
+            elif response.status == LogsQueryStatus.PARTIAL:
+                data = response.partial_data
+                error = response.partial_error
+                self.logger.error("partial_error: {}".format(error))
+            elif response.status == LogsQueryStatus.SUCCESS:
+                data = response.tables
+
+            for table in data:
+                for row in table.rows:
+                    row_dict = {str(col): str(item)
+                                for col, item in zip(table.columns, row)}
+                    result.append(row_dict)
+        except Exception as e:
+            self.logger.error("Exception in fetch logs: {}".format(e))
+
+        return result
+
+    def filter_logs(self, logs, key, value):
+        return value in [d.get(key) for d in logs]
```

Comment:
value in [d.get(key) for d in logs] =>
[d.get(key) for d in logs if value in d.get(key, '')]

Reply:
Updated